images/aws-ocp-snap/include/create_snapshot.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+NAMESPACE=$1
+VOLUME=$2
+
+TMPFILE=$(mktemp /tmp/tmp.XXXXXXXXXXXXX)
+CHKFILE=$(mktemp /tmp/chk.XXXXXXXXXXXXX)
+
+if [ -z "$NAMESPACE" -o "$NAMESPACE" = " " ]; then
+    echo "A Namepace must be provided or select ALL"
+    exit 2
+fi
+case "$NAMESPACE" in
+        "ALL")
+            oc get pv --no-headers 2> $CHKFILE
+            RESULT=$?
+            NOVOL=$(grep "No resources" $CHKFILE | wc -l)
+            if [ ! $RESULT -eq 0 ] || [ $NOVOL -gt 0 ]; then
+                echo "Error while getting EBS volumes information"
+                exit 2
+            else
+                oc describe pv $(oc get pv --no-headers | awk {'print $1'}) | grep VolumeID | cut -d "/" -f 4 > $TMPFILE
+                while read vol
+                do
+                    echo "Creating snapshot for EBS volume " $vol
+                    aws ec2 create-snapshot --volume-id $vol --description "Automted Snapshot by aws-ocp-snap"
+                done < $TMPFILE
+            fi
+            ;;
+        *)
+            if [ -z "$VOLUME" -o "$VOLUME" = " " ]; then
+                echo "A Volume must be provided or select ALL"
+                exit 2
+            fi
+            case "$VOLUME" in
+                    "ALL")
+                        oc get pvc --no-headers -n $NAMESPACE 2> $CHKFILE
+                        RESULT=$?
+                        NOVOL=$(grep "No resources" $CHKFILE | wc -l)
+                        if [ ! $RESULT -eq 0 ] || [ $NOVOL -gt 0 ]; then
+                            echo "Error while getting EBS volumes information"
+                            exit 2
+                        else
+                            oc describe pv $(oc get pvc --no-headers -n $NAMESPACE | awk {'print $3'}) | grep VolumeID | cut -d "/" -f 4 > $TMPFILE
```

Comment:
While this works, I am aways in favor of using built in queries and output formats (gotemplate/jsonpath) instead of the `describe` command as it is not API bound

Reply:
should be able to replace with this:

```
oc get pv $(oc get pvc -n mongodb-statefulset -o jsonpath='{ .items[*].spec.volumeName }') -o jsonpath='{ range .items[*]}{.spec.cinder.volumeID }{"\n"}{end}'
```

replacing "cinder" with whatever to use for amazon.. "ebs" or whatever

Reply:
Thanks, @etsauer. The field for EBS is 'awsElasticBlockStore'.

Done